client/src/common/components/FormDialog.jsx

```diff
@@ -39,14 +42,30 @@ class FormDialog extends Component {
 
   handleDescriptionChange = description => this.setState({ description });
 
-  handleNameChange = name => this.setState({ name });
+  handleNameChange = name => {
+    this.setState({ name }, this.validateName);
+  };
+
+  validateName = () => {
+    const { name } = this.state;
+
+    const errorMessageId = validateWith(value =>
+      validator.isLength(value, { min: 1, max: 32 })
+    )('user.auth.input.email.invalid')(name);
+
+    this.setState({ errorMessageId });
+  };
 
   handleConfirm = () => {
     const { defaultDescription, defaultName, onConfirm } = this.props;
-    const { description, name } = this.state;
+    const { description, name, errorMessageId } = this.state;
+
+    this.validateName();
 
-    if (defaultDescription !== description || defaultName !== name) {
-      name && onConfirm(name, description);
+    if (!errorMessageId) {
```

Comment:
I'm not sure how this upposed to work:
- you take from state `errorMessageId`
- you call `validateName` which set that exxact part of state
- you check in `if` the value of `errorMessageId` which have previous value (before `validateName`)

And additionally `setState` is async so you cannot be sure if this is already set after `validateName` finishes

Reply:
yeah, I was also not 100% sure how this was working as the setState is async.

I've reimplemented the logic, please check it right now